webpack/assets/javascripts/react_app/components/Pagination/Pagination.js

```diff
@@ -2,25 +2,26 @@ import React from 'react';
 import Proptypes from 'prop-types';
 import { isEmpty } from 'lodash';
 import { Paginator } from 'patternfly-react';
-import { getURI, getURIpage, getURIperPage, changeQuery, translatePagination } from './PaginationHelper';
+import { perPageReset, getURI, getURIpage, getURIperPage, changeQuery, translatePagination } from './PaginationHelper';
 import './pagination.scss';
 
 const Pagination = ({ data }) => {
   const urlPage = getURIpage();
-  const urlPerPage = getURIperPage();
+  const currPerPage = getURIperPage() || data.perPage;
   const className = isEmpty(data.classNames) ? 'col-md-12' : `col-md-12 ${data.classNames.pagination_classes}`;
+  const perPageSelectFnc = perPageReset(changeQuery, 'per_page', currPerPage);
 
   return (
       <Paginator
         pagination={{
           page: urlPage || 1,
-          perPage: urlPerPage || data.perPage,
+          perPage: currPerPage,
           perPageOptions: data.perPageOptions,
         }}
         viewType={data.viewType}
         itemCount={data.itemCount}
         onPageSet={page => changeQuery(getURI(), { page })}
-        onPerPageSelect={perPage => changeQuery(getURI(), { per_page: perPage })}
+        onPerPageSelect={perPage => perPageSelectFnc(getURI(), { per_page: perPage })}
```

Comment:
coveralls shows that this line is not covered, I see that you added a unit test for `perPageSelectFnc`, but could we add an integration for this? Pagination is a fundamental feature, so I think it necessary.

Reply:
:thinking:  I don't believe it's necessary because of two points:

* `Paginator` is a very nested component. It contains `PaginationRow` that has `FormGroup` that has `DropdownButton` which is imported from `react-bootsrap`.  `DropdownButton` contains at some level a `div` element. On that praticular element I have to trigger a `select` event and verify that `perPage => perPageSelectFnc(....)` is called.  In other words, this test contains implementation details and if one day `Paginator` will be replaced by other component the test will fail even though it may work. 

* `perPageSelect` unit test checks the behavior rather the integration with `Paginator` which I believe is good enough since we can assume `Paginator` works correctly and when per page is selected the function is called.

If you still think it's necessary, I don't mind working on it. I just wanted to give my thought why I didn't think of doing so. :smile_cat: :coffee:

Reply:
I don't think that's possible either.

I can't catch `onPerPageSelect` because it is been called from another function used in the menu item.

If you believe it's possible, can you please show me how?